doc/specs/#1564 - Settings UI/cascading-settings.md

```diff
@@ -0,0 +1,144 @@
+![Add new profile](./add-new-profile.png)
+
+### Reset Profile button
+
+On the Advanced pivot of a profile's page, there will be a button at the bottom for resetting a profile called "Reset to default settings". This button will remove the user's custom settings inside this profile's object and reset it to defaults, prioritizing profile.defaults then defaults.json.
+
+### "Apply to all profiles" button
+
+A way we could apply settings to all profiles is by adding a "Copy settings to..." button to the Advanced page of each profile. This button will open a content dialog with a tree view listing every profile setting. The user can select which settings they would like to copy over to another profile. At the bottom of the content dialog will list the user's profiles with checkboxes, allowing them to pick which profiles they'd like to copy settings to.
+
+![Copy settings button](./copy-settings-1.png)
+
+![Copy settings modal](./copy-settings-2.png)
+
+## Previously considered ideas
+
+These ideas were considered however we will not be moving forward with them.
+
+### 1: \<inherit\> option
+
+Each setting is an Editable ComboBox (except for boolean and enumerable settings). For booleans, the items will be Enabled and Disabled in a regular ComboBox. Enumerable settings will have their options listed in a regular ComboBox. For integers, most commonly used numbers will be listed.
+
+![Dropdown inheritance](./inheritance-dropdown.png)
+
+**Pros**
+
+- Doesn't clutter the screen.
+
+**Cons**
+
+- Every setting is a dropdown.
+
+**Pitfalls**
+
+- How will color pickers work with this scenario?
+
+**Other considerations**
+
+Each dropdown has either "inherit" or "custom". If the user selects "custom", the original control will appear (i.e. a color picker for colors or a number picker for integers).
+
+This option was not chosen because it added too much overhead for changing a setting. For example, if you wanted to enable acrylic, you'd have to click the dropdown, select custom, watch the checkbox appear, and then select the checkbox.
+
+### 2: Lock Button
+
+Every setting will have a lock button next to it. If the lock is locked, that means the setting is being inherited from Global, and the control is disabled. If the user wants to edit the setting, they can click the lock, which will changed it to the unlocked lock icon, and the control will become enabled.
+
+![Locks inheritance](./inheritance-locks.png)
+
+**Pros**
+
+- Least amount of clutter on the screen while still keeping the original controls
+
+**Cons**
+
+- The lock concept is slightly confusing. Some may assume locking the setting means that it *won't* be inherited from Global and that it's "locked" to the profile. This is the opposite case for its current design. However, flipping the logic of the locks wouldn't make sense with an unlocked lock and a disabled control.
+
+## Capabilities
+
+### Accessibility
+
+All of these additions to the settings UI will have to be accessibility tested.

```

Comment:
I feel like it might be worth a callout that it's not really accessible to anyone to know what the U-turn arrow means without at least a tooltip and at most some sort of tutorial overlay/popup on the first encounter like a video game.

Reply:
I added a note that the reset button will have a tooltip.